rmgpy/rmg/main.py

```diff
@@ -770,6 +771,14 @@ def execute(self, initialize=True, **kwargs):
                     for p in range(reaction_system.n_sims):
                         reactor_done = True
                         objects_to_enlarge = []
+
+                        if self.solvent:
```

Comment:
Is it possible to have this code in base.pyx or liquid.pyx instead of in main?

Reply:
I think it is possible, but I'm not sure if that would be desirable...In that case, RMG will need to load database again in `liquid.pyx` because it needs `solvent_data = self.database.solvation.get_solvent_data(self.solvent)` to calculate viscosity, which will take a longer time in total. I guess another way would be pass in the `self.database` attribute. However, it looks like the current RMG design tries to let `main.py` handle things that require the database information, so I think it is more consistent to keep this code in `main.py`. Let me know your thoughts! @mjohnson541